torch/fx/graph.py

```diff
@@ -101,10 +113,78 @@ def create_node(self, op: str, target: Target,
         self._mark_uses(args)
         self._mark_uses(kwargs)
         n = Node(self, name if name is not None else self._name(target), op, target, args, kwargs)
-        assert self._nodes[-1].op == 'return'
-        self._nodes.insert(-1, n)
+        self.insert_node_before(n=n, before=self._insert_point)
         return n
 
+    def insert_node_before(self, n : Node, before : Node) -> Node:
```

Comment:
How do you get a node that is not already inserted into the graph? I can see `move_node_before` as being a public function, and I can see `create_node` as taking an optional before, but I don't see how this can be used.